src/main/java/mate.academy/Main.java

```diff
@@ -8,9 +8,8 @@
 public class Main {
 
     public static void main(String[] args) {
-        // Please test your Injector here. Feel free to push this class as a part of your solution
         Injector injector = Injector.getInjector();
-        ProductService productService = null;
+        ProductService productService = (ProductService) injector.getInstance(ProductService.class);
         List<Product> products = productService.getAllFromFile("products.txt");
```

Comment:
Ensure that the `getAllFromFile` method properly handles exceptions related to file operations. It should not just throw them up the call stack, but rather handle them gracefully, possibly by logging an error and returning an empty list or a default value.